utils/components.py

```diff
@@ -298,3 +301,302 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+
+    config.load_incluster_config()
+
+    model_endpoint = ""
+    model_name = ""
+    model_api_key = ""
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+
+    with client.ApiClient() as api_client:
+        core_api = client.CoreV1Api(api_client)
+
+        try:
+            secret = core_api.read_namespaced_secret(secret_name, namespace)
+            print(f"Reading secret {secret_name} data...")
+            model_api_key = base64.b64decode(secret.data["api_key"]).decode("utf-8")
+            model_name = base64.b64decode(secret.data["model_name"]).decode("utf-8")
+            model_endpoint = base64.b64decode(secret.data["endpoint"]).decode("utf-8")
+        except (ApiException, KeyError) as e:
+            print(f"""
+            ####################################################### ERROR ################################################################
+            # Error reading {secret_name}. Ensure you created a secret with this name in namespace {namespace} and has 'api_key' present #
+            ##############################################################################################################################
+            """)
+            sys.exit(1)
+
+    request_auth = {"Authorization": f"Bearer {model_api_key}"}
+    request_body = {
+        "model": model_name,
+        "messages": [{"role": "user", "content": "tell me a funny joke."}],
+    }
+    # Make 3 attempts
+    for i in range(1, 3):
+        resp = requests.post(
+            f"{model_endpoint}/chat/completions",
+            headers=request_auth,
+            data=json.dumps(request_body),
+            verify=os.environ["SDG_CA_CERT_PATH"],
+        )
+        if resp.status_code != 200:
+            print(f"Model Server {model_name} is not available. Attempt {i}/3...")
+            time.sleep(30)
+        else:
+            print(f"""
+            ################### INFO #######################
+            # Model Server {model_name} is up and running. #
+            ################################################
+            """)
+            pass
+    print(f"""
+    ############################################ ERROR ####################################################
+    # Model Server {model_name} is unavailable. Ensure the model is up and it is ready to serve requests. #
+    #######################################################################################################
+    """)
+    sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_model_registry(
+    model_registry_endpoint: Optional[str],
+    model_name: Optional[str],
+    model_version: Optional[str],
+):
+    import sys
+
+    from model_registry import ModelRegistry
+
+    try:
+        model_registry = ModelRegistry(
+            server_address=model_registry_endpoint,
+            author="ilab pipeline",
```

Comment:
Not that I think it matters much
```suggestion
            author="InstructLab Pipeline",
```